scripts/ISM/ISM.py

```diff
@@ -0,0 +1,263 @@
+        return include_controls
+
+    def write_catalog(self, catalogs_path, ism_name):
+        """Wrap and write oscal catalog object."""
+        ism_dir_path = catalogs_path / ism_name
+        ism_dir_path.mkdir(exist_ok=True)
+        ism_file_path = ism_dir_path / 'catalog.json'
+        self._ism_catalog.oscal_write(ism_file_path)
+
+    def write_profiles(self, profiles_dir, ism_name):
+        """Write out all profiles."""
+        for security_level in self._profiles.keys():
+            profile_dir = profiles_dir / (ism_name + '_' + security_level)
+            profile_dir.mkdir(exist_ok=True)
+            profile_path = profile_dir / 'profile.json'
+            self._profiles[security_level].oscal_write(profile_path)
+
+
+class ISM(Command):
+    """
+    Convert the Australian goverment information security manual (in various versions) into catalogs and profiles.
+
+    This CLI has presumptions on resource structures that are returned.
+
+    Please note that this project current presumes information about the project structure.
+    """
+
+    def _init_arguments(self):
+        self.add_argument('-r', '--root-dir', help='Trestle project root.', default='./')
+
+    def _run(self, args):
+        # little test
+        root_dir = pathlib.Path(args.root_dir).resolve()
+        catalogs_dir = root_dir.joinpath('catalogs').resolve()
+        profiles_dir = root_dir.joinpath('profiles').resolve()
+        cache_dir = root_dir.joinpath('non-oscal-data').resolve()
+        ism_json_file = root_dir.joinpath('scripts/ISM/ism_editions.json').resolve()
+
+        if not root_dir.exists():
+            logger.error('Root trestle project does not exist')
+            return 1
+        if not catalogs_dir.exists():
+            logger.error('Catalogs directory does not exist.')
+            return 1
+
+        if not profiles_dir.exists():
+            logger.error('Profiles directory does not exist.')
+            return 1
+
+        if not cache_dir.exists():
+            logger.error('Cache directory does not exist.')
+            return 1
+
+        ism_versions = json.load(ism_json_file.open())
+
+        for ism_file in ism_versions['isms']:
+            # ISM file format: 'ISM - List of Security Controls (August 2019).xml'
+            logger.info(ism_file)
+            url = ism_file['version_url']
+            ism_manager = ISMManager()
+            ism_manager.fetch_ism(url)
+            revision_date = ism_file['version_name'].split()
+            revision_string = revision_date[0] + '_' + revision_date[1]
+            logger.info(f'Revision date: {revision_date}')
+            logger.info(f'Revision string: {revision_string}')
+            logger.info(revision_string)
+            ism_name = 'ISM_' + revision_string
+
+            ism_manager.create_ism_catalogs(revision_string)
+            # This is presumed to be relative for now to the catalog repo based on this
+            ism_manager.write_catalog(catalogs_dir, ism_name)
```

Comment:
plural catalogs would be more consistent here

Reply:
Changed the names to be consistent.